src/transformers/generation/logits_process.py

```diff
@@ -1330,8 +1330,12 @@ def __call__(self, input_ids: torch.LongTensor, scores: torch.FloatTensor) -> to
 
 class WhisperTimeStampLogitsProcessor(LogitsProcessor):
     r"""
-    Whisper specific Processor. This processor can be used to force a list of tokens. The processor will set their log
-    probs to `inf` so that they are sampled at their corresponding index.
+
+    [`LogitsProcessor`] that modifies the logits for the generation of timestamps in the transcription. When the input
+    tokens are at a specific threshold, the processor sets the scores to negative infinity and returned. The timestamps
+    are processed to be in pairs except right before the end-of-sequence token. If the total probability of all
+    timestamp tokens is greater than any individual non-timestamp token, the processor sets those non-timestamp logits
```

Comment:
```suggestion
    [`LogitsProcessor`] that modifies the logits for the generation of timestamps in the transcription. When the input
    tokens are at a specific threshold, the processor sets the scores to negative infinity. The processor makes sure that timestamp tokens appear in pair, by doing .... This is done because ......
     It also ensure that when  the predicted probability of sampling any of the timestamp token is greater than ........ This is done to ensure that..... 
```